Author: Boris Pek <tehnick-8@yandex.ru>
Date:   Thu Jun 28 00:39:42 2018 +0300

    Make build of test application optional and disable it

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -18,9 +18,11 @@
 include(ECMGeneratePriFile)
 
 option(BUILD_WITH_QT4 "Build qtkeychain with Qt4 no matter if Qt5 was found" OFF)
+option(BUILD_TEST_APPLICATION "Build test application" OFF)
 option(BUILD_TRANSLATIONS "Build translations" ON)
 option(QTKEYCHAIN_STATIC "Build static library" OFF)
 
+
 if(CMAKE_SYSTEM_NAME STREQUAL Android)
     set(ANDROID 1)
 endif()
@@ -224,8 +226,10 @@
   ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
 )
 
-add_executable( testclient testclient.cpp )
-target_link_libraries( testclient ${QTKEYCHAIN_TARGET_NAME})
+if(BUILD_TEST_APPLICATION)
+    add_executable( testclient testclient.cpp )
+    target_link_libraries( testclient ${QTKEYCHAIN_TARGET_NAME})
+endif()
 
 
 ###
