commit 5a9fecfcf6bfeecb7be1013c5534b3057f8b1232
Author: Boris Pek <tehnick-8@yandex.ru>
Date:   Sun Jul 1 03:13:39 2018 +0300

    Fix cross-compilation

diff --git a/Source/WebKit/qt/WidgetApi/qgraphicswebview.h b/Source/WebKit/qt/WidgetApi/qgraphicswebview.h
index 1aa5b66c..ead6683f 100644
--- a/Source/WebKit/qt/WidgetApi/qgraphicswebview.h
+++ b/Source/WebKit/qt/WidgetApi/qgraphicswebview.h
@@ -20,8 +20,8 @@
 #ifndef QGraphicsWebView_h
 #define QGraphicsWebView_h
 
-#include <QtWebKit/qwebkitglobal.h>
-#include <QtWebKitWidgets/qwebpage.h>
+#include <qwebkitglobal.h>
+#include <qwebpage.h>
 #include <QtCore/qurl.h>
 #include <QtGui/qevent.h>
 #include <QtGui/qicon.h>
diff --git a/Source/WebKit/qt/WidgetApi/qwebframe.h b/Source/WebKit/qt/WidgetApi/qwebframe.h
index 1c3cb07c..7997764c 100644
--- a/Source/WebKit/qt/WidgetApi/qwebframe.h
+++ b/Source/WebKit/qt/WidgetApi/qwebframe.h
@@ -26,7 +26,7 @@
 #include <QtCore/qvariant.h>
 #include <QtGui/qicon.h>
 #include <QtNetwork/qnetworkaccessmanager.h>
-#include <QtWebKit/qwebkitglobal.h>
+#include <qwebkitglobal.h>
 
 QT_BEGIN_NAMESPACE
 class QRect;
diff --git a/Source/WebKit/qt/WidgetApi/qwebinspector.h b/Source/WebKit/qt/WidgetApi/qwebinspector.h
index c5f097e2..9dfa22d8 100644
--- a/Source/WebKit/qt/WidgetApi/qwebinspector.h
+++ b/Source/WebKit/qt/WidgetApi/qwebinspector.h
@@ -20,10 +20,10 @@
 #ifndef QWEBINSPECTOR_H
 #define QWEBINSPECTOR_H
 
-#include <QtWebKit/qwebkitglobal.h>
-#include <QtWebKitWidgets/qwebpage.h>
+#include <qwebkitglobal.h>
+#include <qwebpage.h>
 
-#include <QtWebKitWidgets/qwebview.h>
+#include <qwebview.h>
 
 class QWebInspectorPrivate;
 
diff --git a/Source/WebKit/qt/WidgetApi/qwebpage.h b/Source/WebKit/qt/WidgetApi/qwebpage.h
index 822bd219..4972858a 100644
--- a/Source/WebKit/qt/WidgetApi/qwebpage.h
+++ b/Source/WebKit/qt/WidgetApi/qwebpage.h
@@ -21,8 +21,8 @@
 #ifndef QWEBPAGE_H
 #define QWEBPAGE_H
 
-#include <QtWebKit/qwebkitglobal.h>
-#include <QtWebKit/qwebsettings.h>
+#include <qwebkitglobal.h>
+#include <qwebsettings.h>
 
 #include <QtCore/qobject.h>
 #include <QtCore/qurl.h>
diff --git a/Source/WebKit/qt/WidgetApi/qwebview.h b/Source/WebKit/qt/WidgetApi/qwebview.h
index 0ed324da..c7d346da 100644
--- a/Source/WebKit/qt/WidgetApi/qwebview.h
+++ b/Source/WebKit/qt/WidgetApi/qwebview.h
@@ -21,8 +21,8 @@
 #ifndef QWEBVIEW_H
 #define QWEBVIEW_H
 
-#include <QtWebKit/qwebkitglobal.h>
-#include <QtWebKitWidgets/qwebpage.h>
+#include <qwebkitglobal.h>
+#include <qwebpage.h>
 #include <QtCore/qurl.h>
 #include <QtGui/qicon.h>
 #include <QtGui/qpainter.h>
diff --git a/Source/WebKit/qt/declarative/experimental/plugin.cpp b/Source/WebKit/qt/declarative/experimental/plugin.cpp
index 66343af7..6071a9d5 100644
--- a/Source/WebKit/qt/declarative/experimental/plugin.cpp
+++ b/Source/WebKit/qt/declarative/experimental/plugin.cpp
@@ -19,16 +19,16 @@
 
 #include "config.h"
 
-#include "private/qquicknetworkreply_p.h"
-#include "private/qquicknetworkrequest_p.h"
-#include "private/qquickwebpage_p.h"
-#include "private/qquickwebview_p.h"
-#include "private/qtwebsecurityorigin_p.h"
-#include "private/qwebdownloaditem_p.h"
-#include "private/qwebkittest_p.h"
-#include "private/qwebnavigationhistory_p.h"
-#include "private/qwebpermissionrequest_p.h"
-#include "private/qwebpreferences_p.h"
+#include "qquicknetworkreply_p.h"
+#include "qquicknetworkrequest_p.h"
+#include "qquickwebpage_p.h"
+#include "qquickwebview_p.h"
+#include "qtwebsecurityorigin_p.h"
+#include "qwebdownloaditem_p.h"
+#include "qwebkittest_p.h"
+#include "qwebnavigationhistory_p.h"
+#include "qwebpermissionrequest_p.h"
+#include "qwebpreferences_p.h"
 
 #include <QtQml/qqml.h>
 #include <QtQml/qqmlextensionplugin.h>
diff --git a/Source/WebKit/qt/declarative/plugin.cpp b/Source/WebKit/qt/declarative/plugin.cpp
index 2ffc8ea0..d3ce62b7 100644
--- a/Source/WebKit/qt/declarative/plugin.cpp
+++ b/Source/WebKit/qt/declarative/plugin.cpp
@@ -25,11 +25,11 @@
 #include <QtQml/qqmlextensionplugin.h>
 
 #if defined(HAVE_WEBKIT2)
-#include "private/qquickwebpage_p.h"
-#include "private/qquickwebview_p.h"
-#include "private/qwebiconimageprovider_p.h"
-#include "private/qwebloadrequest_p.h"
-#include "private/qwebnavigationrequest_p.h"
+#include "qquickwebpage_p.h"
+#include "qquickwebview_p.h"
+#include "qwebiconimageprovider_p.h"
+#include "qwebloadrequest_p.h"
+#include "qwebnavigationrequest_p.h"
 
 #include <QtNetwork/qnetworkreply.h>
 #include <QtQml/qqmlengine.h>
diff --git a/Source/WebKit/qt/declarative/public.pri b/Source/WebKit/qt/declarative/public.pri
index 6c3c2d40..ba0100d5 100644
--- a/Source/WebKit/qt/declarative/public.pri
+++ b/Source/WebKit/qt/declarative/public.pri
@@ -35,6 +35,10 @@ SOURCES += plugin.cpp
 build?(webkit2): {
     DEFINES += HAVE_WEBKIT2
     QT += network
+
+    INCLUDEPATH += \
+        $$ROOT_WEBKIT_DIR/Source/WebKit/qt/Api \
+        $$ROOT_WEBKIT_DIR/Source/WebKit2/UIProcess/API/qt
 }
 
 
diff --git a/Source/WebKit2/UIProcess/API/qt/tests/qmltests/tst_qmltests.cpp b/Source/WebKit2/UIProcess/API/qt/tests/qmltests/tst_qmltests.cpp
index ab57556b..4d222b99 100644
--- a/Source/WebKit2/UIProcess/API/qt/tests/qmltests/tst_qmltests.cpp
+++ b/Source/WebKit2/UIProcess/API/qt/tests/qmltests/tst_qmltests.cpp
@@ -20,7 +20,7 @@
 #include "../bytearraytestdata.h"
 #include "../util.h"
 
-#include "private/qquickwebview_p.h"
+#include "qquickwebview_p.h"
 #include <QGuiApplication>
 #include <QVarLengthArray>
 #include <QtQuickTest/quicktest.h>
diff --git a/Source/WebKit2/UIProcess/API/qt/tests/util.cpp b/Source/WebKit2/UIProcess/API/qt/tests/util.cpp
index 3b0c45f7..146b1830 100644
--- a/Source/WebKit2/UIProcess/API/qt/tests/util.cpp
+++ b/Source/WebKit2/UIProcess/API/qt/tests/util.cpp
@@ -23,8 +23,8 @@
 #include <stdio.h>
 
 #if defined(HAVE_QTQUICK) && HAVE_QTQUICK
-#include "private/qquickwebview_p.h"
-#include "private/qwebloadrequest_p.h"
+#include "qquickwebview_p.h"
+#include "qwebloadrequest_p.h"
 #endif
 
 void addQtWebProcessToPath()
diff --git a/Tools/MiniBrowser/qt/BrowserWindow.cpp b/Tools/MiniBrowser/qt/BrowserWindow.cpp
index 8472d5cf..72478d16 100644
--- a/Tools/MiniBrowser/qt/BrowserWindow.cpp
+++ b/Tools/MiniBrowser/qt/BrowserWindow.cpp
@@ -28,8 +28,8 @@
 
 #include "BrowserWindow.h"
 
-#include "private/qquickwebpage_p.h"
-#include "private/qquickwebview_p.h"
+#include "qquickwebpage_p.h"
+#include "qquickwebview_p.h"
 #include "utils.h"
 
 #include <QQmlEngine>
diff --git a/Tools/MiniBrowser/qt/MiniBrowserApplication.cpp b/Tools/MiniBrowser/qt/MiniBrowserApplication.cpp
index f2eadba9..f6613ee7 100644
--- a/Tools/MiniBrowser/qt/MiniBrowserApplication.cpp
+++ b/Tools/MiniBrowser/qt/MiniBrowserApplication.cpp
@@ -35,7 +35,7 @@
 #if HAVE(QTTESTSUPPORT)
 #include "QtTestSupport.h"
 #endif
-#include "private/qquickwebview_p.h"
+#include "qquickwebview_p.h"
 #include "utils.h"
 #include <QRegExp>
 #include <QEvent>
diff --git a/Tools/MiniBrowser/qt/UrlLoader.cpp b/Tools/MiniBrowser/qt/UrlLoader.cpp
index a0658e98..4ec670d0 100644
--- a/Tools/MiniBrowser/qt/UrlLoader.cpp
+++ b/Tools/MiniBrowser/qt/UrlLoader.cpp
@@ -27,8 +27,8 @@
  */
 
 #include "UrlLoader.h"
-#include "private/qquickwebview_p.h"
-#include "private/qwebloadrequest_p.h"
+#include "qquickwebview_p.h"
+#include "qwebloadrequest_p.h"
 
 #include <QDebug>
 #include <QFile>
